src/main/java/io/reactivesocket/ReactiveSocket.java

```diff
@@ -95,6 +95,13 @@ public void error(Throwable e) {
     void onRequestReady(Completable c);
 
     /**
+     * Invoked when ReactiveSocket is closed
+     *
+     * @throws IllegalStateException if this method is called more than once per instance
+     */
+    void onShutdown(Completable c);
```

Comment:
Actually, why to ask for a `Completable`? This is like a close future for the socket, which is a single instance that anyone can subscriber/listen to. So, why not have something like:

```java
Completable shutdownListener();
```